### PR TITLE
[CARBONDATA-202] Handled exception thrown in beeline for all dictionary 

### PR DESCRIPTION
**Pull request by foryou2030**
# Why raise this pr?
Exception thrown in Beeline for data loading when dictionary file content is not in correct format is not proper.
# How to solve?
use accumulator as flag, record the bad record in dictionary files, then throw exception

### CODE REVIEW

**Comment by foryou2030** [comment]
CI passed
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/92/

**Comment by foryou2030** [comment]
CI passed
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/95/

File: integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 1]
```diff
@@ -588,30 +588,59 @@ object GlobalDictionaryUtil extends Logging {
...
         (columnName, value)
-      })
+      }
 
+      val accumulator = sqlContext.sparkContext.accumulator(0)
+      // read local dictionary file, and spilt (columnIndex, columnValue)
+      val basicRdd = sqlContext.sparkContext.textFile(allDictionaryPath)
+        .map(x => parseRecord(x, accumulator)).persist()
+      // for accumulator updates performed inside actions only
+      basicRdd.count()
```
Comment (sujith71955) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: again we are calling action which is not necessary

File: integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 2]
```diff
@@ -588,30 +588,59 @@ object GlobalDictionaryUtil extends Logging {
...
+            case ex: Exception =>
+              logError("Read a bad dictionary record: " + x)
+              accum += 1
+          }
         }
         (columnName, value)
-      })
+      }
 
+      val accumulator = sqlContext.sparkContext.accumulator(0)
```
Comment (sujith71955) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: return the accumulator by updating and let caller validate and throw exception

Comment (foryou2030) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: fixed

File: integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala (hunk 1 above)
Comment (foryou2030) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: fixed

File: integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 3]
```diff
@@ -791,16 +819,23 @@ object GlobalDictionaryUtil extends Logging {
...
             // read exist dictionary and combine
             val inputRDD = new CarbonAllDictionaryCombineRDD(allDictionaryRdd, model)
               .partitionBy(new ColumnPartitioner(model.primDimensions.length))
             // generate global dictionary files
             val statusList = new CarbonGlobalDictionaryGenerateRDD(inputRDD, model).collect()
             // check result status
             checkStatus(carbonLoadModel, sqlContext, model, statusList)
+            // if the dictionary contains wrong format record, throw ex
+            if (accumulator.value > 0) {
+              throw new DataLoadingException("Data Loading failure, the dictionary file " +
```
Comment (sujith71955) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: i think content term will be too generic, we can say dictionary values, Whats your opinion? better to use more meaningful term which is related to that particular file, like Dictionary values

Comment (foryou2030) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: u r right. handled

**Comment by foryou2030** [comment]
CI passed
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/97/

**Comment by sujith71955** [comment]
LGTM

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**